src/components/Landing/ApplicantActionsDisplay.js

```diff
@@ -0,0 +1,177 @@
+import React, { Component } from "react";
+import { navigate } from "gatsby";
+import { compose } from "recompose";
+
+import Row from "react-bootstrap/Row";
+
+import ActionCard from "../Landing/ActionCard";
+import Loader from "../Loader";
+
+import { AuthUserContext, withFirebase } from "upe-react-components";
+import { withSettings } from "../API/SettingsContext";
+
+import { Wrapper, Title } from "../../styles/global";
+
+class ApplicantActionsDisplay extends Component {
+  _initFirebase = false;
+  state = {
+    loading: true,
+    application: null,
+  };
+  static contextType = AuthUserContext;
+
+  componentDidMount() {
+    if (this.props.firebase && !this._initFirebase) this.loadData();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.firebase && !this._initFirebase) this.loadData();
+    if (typeof window !== "undefined") {
+      import("bs-custom-file-input").then((bsCustomFileInput) => {
+        bsCustomFileInput.init();
+      });
+    }
+  }
+
+  loadData = async () => {
+    this._initFirebase = true;
+
+    this.props.firebase
+      .application(!!this.context ? this.context.uid : "1")
+      .get()
+      .then((snapshot) => {
+        if (!snapshot.exists)
+          this.setState({ application: null, loading: false });
+        else this.setState({ application: snapshot.data(), loading: false });
+      })
+      .catch(() => {
+        this.setState({ application: null, loading: false });
+        console.log("No application");
+      });
+  };
+
+  navigateTo = (page) => {
+    navigate(page);
+  };
+
+  render() {
+    const { loading, application } = this.state;
+
+    // Data Loading
+    if (loading) return <Loader />;
+
+    // Check if it's interview Day
+    const date = new Date();
+    let isDay = false;
+    this.props.settings.timeslotDays.forEach((day) => {
+      const transformedDay = day.toDate();
+      isDay |=
+        transformedDay.getDay() === date.getDay() &&
+        transformedDay.getMonth() === date.getMonth() &&
+        transformedDay.getYear() === date.getYear();
+    });
+
+    return (
+      <>
+        <Title>
+          <h1> Actions </h1>
+        </Title>
+        <Wrapper
+          paddingRight={"15%"}
+          paddingLeft={"15%"}
+          paddingTop={"80px"}
+          paddingBottom={"100px"}
+        >
+          <Row
+            lg={2}
+            md={2}
+            sm={1}
+            xl={2}
+            xs={1}
+            style={{ alignItems: "center", justifyContent: "center" }}
+          >
+            {!this.context && (
```

Comment:
The pattern of checking some variable to render the action card is repeated a lot, although I'm not entirely sure how to fix it. You could move that logic to inside the action card, e.g. take context and settings as props, as well as a function to define what to check from context and settings, like withAuthorization, but there may still be a similar amount of repeated code? Not sure, but I think there should be a way of simplifying this